### ISSUE TITLE
Authentication: Add OIDC Integration

### ISSUE THREAD

**Issue by redxtech** [wger-project/wger#1797]
## Use case

I would like to see OIDC support for authentication, to make it easier to integrate with existing setups that use auth providers such as authentik.

## Proposal

Django appears to have support for OIDC build in (https://django-oidc-provider.readthedocs.io/en/master/), so adding it here should definitely be easier than if you were to start from scratch. If that somehow doesn't apply here, there should probably be another way to implement this as well. Let me know if I'm wrong!

**Comment by kmoy1** [comment]
Would like to work on this

**Comment by kmoy1** [comment]
Is part of this request to add an OIDC authentication backend and authenticate against that instead of the default `django.contrib.auth.backends.ModelBackend` (checking against the User model)? If that's the case, I am not sure `django-oidc-provider` provides one- may need to use the [mozilla-django-oidc](https://mozilla-django-oidc.readthedocs.io/en/stable/installation.html?highlight=authentication_backend) library instead which does.

**Comment by rolandgeider** [comment]
We definitely need a user object for the rest of the application to work. I'm not sure how the oidc provider works, but I assumed it would do that automatically

**Comment by kmoy1** [comment]
Yes, I was just thinking that if we set up an OIDC IDP like Authentik, we would manage user accounts there instead of locally in database.sqlite3 like we do currently. Unless I'm mistaken?

**Comment by kmoy1** [comment]
I've added a basic example of support for OIDC authentication using the `mozilla-django-oidc` library. However, this CR does require the manual setup of an OIDC provider (I used Authentik here). I didn't see another way around this.

**Comment by InputObject2** [comment]
If we're setting up oidc auth, the oidc provider part isn't wger's responsibility. When you say "require the manual setup" did you mean we can no longer use "local" auth?

Also needing a couple environment variables is pretty common. These are well-known variables that most providers use, so it wouldn't be limited to working with Authentik.

**Comment by rolandgeider** [comment]
This only allows to setup one OIDC provider right? Is that enough or would we potentially want more?

**Comment by kmoy1** [comment]
Yes, this was meant specifically for the Authentik. But the mozilla-django-oidc library can definitely extend to other OIDC providers too. Again, the issue is that (as far as I know) the provider needs to be configured manually and separately from this app.

**Comment by LeVraiRoiDHyrule** [comment]
Hi, is there news on this ? Auth headers works right now but true OIDC would be even better.

**Comment by Keridos** [comment]
I currently host wger on my truenas scale system as an app. It seems auth headers are broken on that. While technically not this projects fault, I would love to see OIDC support. Currently the chain of systems when using forwardAuth is:

wger -> reverse-proxy -> authelia -> ldap proxy -> authentik 

WIth OIDC it would be much simpler:
wger <-> Authentik (or keycloak, google etc.)

**Comment by jsparrowio** [comment]
+1 on this. Self hosted SSO apps like Authentik support would be great and much more secure then RP SSO.

**Comment by rolandgeider** [comment]
OIDC integration can be configured by saving a couple of entries to the db 🥳  Check [the docs](https://wger.readthedocs.io/en/latest/administration/social_auth.html#generic-openid-connect-oidc) for details.

Closing here, but if something doesn't quite work or is unclear, please re-open
